client/src/shapes/shapes/rect/hitbox.ts

```diff
@@ -40,67 +42,112 @@ export const rectHitbox = (rectangle: RectSchema) => (point: Coordinate) => {
     );
   }
 
-  const radius = Math.min(
-    borderRadius,
-    normalizedWidth / 2,
-    normalizedHeight / 2,
-  );
+  let radii: number[];
+  if (typeof borderRadius === 'number') {
+    radii = [borderRadius, borderRadius, borderRadius, borderRadius];
+  } else {
+    radii = borderRadius;
+  }
 
-  const verticalWidth = Math.max(normalizedWidth - 2 * radius, 0);
-  const horizontalHeight = Math.max(normalizedHeight - 2 * radius, 0);
+  const maxRadius = Math.min(normalizedWidth / 2, normalizedHeight / 2);
+  const [topLeftRadius, topRightRadius, bottomRightRadius, bottomLeftRadius] =
+    radii.map((r) => Math.min(Math.max(r, 0), maxRadius));
 
-  const rectVertical = rectHitbox({
-    ...rectangle,
-    at: { x: x + radius, y },
-    width: verticalWidth,
-    height: normalizedHeight,
-    borderRadius: 0,
-    rotation: 0,
-    stroke,
-  });
+  const rectangles = [
```

Comment:
can this get a type annotation so that if there is a type error it appears here instead of later on when its being used